xbmc/cores/VideoPlayer/VideoRenderers/HwDecRender/DXVAHD.cpp

```diff
@@ -1029,3 +1057,127 @@ DXGI_COLOR_SPACE_TYPE CProcessorHD::AvToDxgiColorSpace(const DXGIColorSpaceArgs&
 
   return DXGI_COLOR_SPACE_YCBCR_STUDIO_G22_LEFT_P709;
 }
+
+bool CProcessorHD::IsSuperResolutionSuitable(const VideoPicture& picture)
+{
+  if (picture.iWidth > 1920)
```

Comment:
Is there a scaling limit for good results? like 2x the original resolution, 4x or ...?

Reply:
There is no specific limit in percentage (theoretically it is possible to upscale 320p to 4K) but Nvidia says the maximum input resolution is 1440p and the output is 5K.

See this FAQ for more details:
https://nvidia.custhelp.com/app/answers/detail/a_id/5448

For convenience and simplification I limited to 1080p and anyway from 1080p to 1440p I think it would be a residual usage and an even higher energy consumption for an insignificant result.

Reply:
Makes sense. No need for a low resolution limit then, less than 360p is pathologic these days.